spikeinterface/preprocessing/phase_shift.py

```diff
@@ -49,15 +49,17 @@ def __init__(self, recording, margin_ms=40.,  inter_sample_shift=None, dtype=Non
         
         force_dtype_back = None
         if dtype is None:
-            if recording.get_dtype().kind in ('i', 'u'):
-                # because of the tapper on margin we need to force the computation to float32
-                # and then go back to the original buffer unless dtype is explicitly forced
-                dtype = np.dtype('float32')
-                force_dtype_back = recording.get_dtype()
- 
+            dtype = recording.get_dtype()
+        # because of the tapper on margin we need to force the computation to float32
+        # and then go back to the original buffer unless dtype is explicitly forced
+        tmp_dtype = np.dtype('float32') if dtype.kind in ('i', 'u') else dtype
```

Comment:
It is better to to put None instead dtype so it avoid the astype() in the get_trace()